### PR TITLE
Add #[Template()] attribute

### LINKED ISSUE #16969: [TwigBridge] Add `#[Template()]` to describe how to render arrays retur…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#46906
| PR author(s) | @nicolas-grekas
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #16969.

### CODE REVIEW

File: components/http_kernel.rst [hunk 1]
```diff
@@ -409,12 +409,12 @@ return a ``Response``.
...
-    There is no default listener inside the Symfony Framework for the ``kernel.view``
-    event. However, `SensioFrameworkExtraBundle`_ *does* add a listener to this
-    event. If your controller returns an array, and you place the `@Template`_
-    annotation above the controller, then this listener renders a template,
-    passes the array you returned from your controller to that template, and
-    creates a ``Response`` containing the returned content from that template.
+    There is a default listener inside the Symfony Framework for the ``kernel.view``
+    event. If your controller returns an array, and you place the
+    :ref:`#[Template()] attribute <templates-template-attribute>` above the
+    controller, then this listener renders a template, passes the array you
```
Comment (94noni) [inline components/http_kernel.rst]:
```suggestion
    controller action, then this listener renders a template, passes the array you
```

File: templates.rst [hunk 2]
```diff
@@ -470,6 +470,38 @@ If your controller does not extend from ``AbstractController``, you'll need to
...
+    // src/Controller/ProductController.php
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\HttpFoundation\Response;
+
+    class ProductController extends AbstractController
+    {
+        #[Template('product/index.html.twig')]
+        public function index(): Response
```
Comment (94noni) [inline templates.rst]: does the return typehint works here?

File: components/http_kernel.rst [hunk 3]
```diff
@@ -409,12 +409,12 @@ return a ``Response``.
...
 .. sidebar:: ``kernel.view`` in the Symfony Framework
 
-    There is no default listener inside the Symfony Framework for the ``kernel.view``
-    event. However, `SensioFrameworkExtraBundle`_ *does* add a listener to this
-    event. If your controller returns an array, and you place the `@Template`_
-    annotation above the controller, then this listener renders a template,
-    passes the array you returned from your controller to that template, and
-    creates a ``Response`` containing the returned content from that template.
+    There is a default listener inside the Symfony Framework for the ``kernel.view``
+    event. If your controller returns an array, and you place the
```
Comment (94noni) [inline components/http_kernel.rst]:
```suggestion
    event. If your controller action returns an array, and you place the
```

wdyt? while we are editing this paragraph, as it is a controller's action that has the attribute

File: templates.rst (hunk 2 above)
Comment (javiereguiluz) [reply templates.rst]: Good catch!

**Comment by javiereguiluz** [comment]
Antoine, thanks for your review.

**Review by 94noni — APPROVED** [review]
my pleasure 👍🏻

**Pull request merged**